app/lib/methods/helpers/debounce.ts

```diff
@@ -21,3 +23,7 @@ export function debounce(func: Function, wait?: number, immediate?: boolean) {
 	_debounce.stop = () => clearTimeout(timeout!);
 	return _debounce;
 }
+
+export function useDebounce(func: Function, wait?: number, immediate?: boolean): (...args: any[]) => void {
```

Comment:
We discussed this internally, but we should consider using a lib instead of creating our own.

My main concern is related to the useCallback usage and here's why https://github.com/xnimorz/use-debounce/blob/8020962aed8038c065f9468fd7cf1d9a1751ce62/src/useDebouncedCallback.ts#L150